src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py

```diff
@@ -503,6 +553,198 @@ def show_release_directives(
                 )
             return cursor.fetchall()
 
+    def get_existing_app_info(self, name: str, role: str) -> dict | None:
```

Comment:
why is this different than show_specific_object?

Reply:
The goal was to also move `show_specific_object` calls to the facade eventually, this is one of those instances, see https://github.com/snowflakedb/snowflake-cli/pull/1870#discussion_r1858668505 for initial context